packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -115,15 +109,18 @@ function getPackageNameWithSuffix(
 }
 
 // Builds the app and runs it on a connected emulator / device.
-function buildAndRun(args: Flags) {
-  process.chdir(path.join(args.root, 'android'));
-  const cmd = process.platform.startsWith('win') ? 'gradlew.bat' : './gradlew';
+function buildAndRun(args: Flags, config: Config) {
+  const androidConfig = config.project.android;
+  if (!androidConfig) {
+    return;
+  }
+  process.chdir(androidConfig.sourceDir);
+
+  const gradlew = process.platform === 'win32' ? 'gradlew.bat' : './gradlew';
```

Comment:
If we remove `process.cwd`, I guess this should be `./android/gradlew`?

Reply:
Yea not quite, because Gradle will think that cwd is `./` and will try to apply tasks to the root project instead of app project. We could, however, pass cwd to `execa` and avoid process.chdir if you like. I left it, because we have similar logic for iOS